app/services/sensor_logs/weather_api_simulation_service.rb

```diff
@@ -0,0 +1,171 @@
+require "net/http"
+require "json"
+
+module SensorLogs
+  class WeatherApiSimulationService
+    def initialize
+      @api_key  = ENV["WEATHERAPI_API_KEY"]
+      @base_url = Settings.weatherapi_url
+    end
+
+    def run
+      if @api_key.blank?
+        Rails.logger.warn(
+          I18n.t("weather_api_simulation.missing_api_key")
+        )
+        return
+      end
+
+      zones = load_target_zones
+      if zones.empty?
+        Rails.logger.info(
+          I18n.t("weather_api_simulation.no_zones_found")
+        )
+        return
+      end
+
+      zones.each{|zone| process_zone(zone)}
+    rescue StandardError => e
+      Rails.logger.error(
+        I18n.t("weather_api_simulation.unexpected_error", error: e.message)
+      )
+    end
+
+    private
+
+    def load_target_zones
+      Zone.includes(:sensors)
+          .where.not(city: nil)
+          .or(Zone.where.not(latitude: nil, longitude: nil))
+          .where(sensors: {status: 0}) # 0 = active
```

Comment:
```suggestion
          .where(sensors: {status: :active})
```
enum có hỗ trợ filter theo string